vispy/visuals/mesh.py

```diff
@@ -321,13 +386,17 @@ def _update_data(self):
                 self._normals.set_data(np.zeros((0, 3), dtype=np.float32))
             self._index_buffer = None
             if md.has_vertex_color():
-                self._colors.set_data(md.get_vertex_colors(indexed='faces'),
-                                      convert=True)
+                colors = md.get_vertex_colors(indexed='faces')
+                colors = colors.astype(np.float32)
             elif md.has_face_color():
-                self._colors.set_data(md.get_face_colors(indexed='faces'),
-                                      convert=True)
+                colors = md.get_face_colors(indexed='faces')
+                colors = colors.astype(np.float32)
+            elif md.has_vertex_value():
+                colors = md.get_vertex_values(indexed='faces')
+                colors = colors.ravel()[:, np.newaxis]
+                colors = colors.astype(np.float32)
```

Comment:
oh, and of course we need to add a conditional that causes it to actually set `colors` when the vertex-value mode is available